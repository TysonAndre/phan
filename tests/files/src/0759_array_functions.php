<?php
function expect_array151(array $x) {
    var_export($x);
}
/**
 * @param mixed $b
 */
function test151($a, $b) {
    expect_array151(array_merge(array_values($a), $b));
    expect_array151(array_keys($a));
    expect_array151(array_keys($a, 'search', true));
    $keys = array_keys($a);
    if (is_string(key($keys))) {  // should warn
        echo "Impossible\n";
    }
    $values = array_values($a);
    $key_t = key($values);

    if (is_string(key($values))) {
        echo "Impossible\n";
    }
    if (key($a) === false) {
        echo "Impossible\n";
    }
}
test151(['abc'], ['xyz', 'def']);
