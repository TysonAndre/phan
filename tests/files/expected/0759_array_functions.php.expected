%s:13 PhanImpossibleCondition Impossible attempt to cast key($keys) of type int|null to string
%s:17 PhanUnusedVariable Unused definition of variable $key_t
%s:19 PhanImpossibleCondition Impossible attempt to cast key($values) of type int|null to string
%s:22 PhanImpossibleTypeComparison Impossible attempt to check if key($a) of type int|null|string is identical to false of type false
